core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -130,6 +130,12 @@ sealed trait Cause[+E] extends Product with Serializable { self =>
       case (_, Interrupt(_))    => false
     }
 
+  final def isTopLevelSilent: Boolean =
```

Comment:
I think we need to do more here. Because if we have, say `Both(cause, silent)` we want to report `cause` but not `silent`. So I think we want to walk the cause tree recursively and eliminate the children of silent nodes. Per my comment below, I think it might be nice to have `silent` have the same compositional semantics as `stackless` where an inner cause can say it doesn't want to be silent. Either way, think this is really something like:

```scala
final def stripSilent: Option[Cause[E]]
```